File: level_1/소수 만들기.py
from itertools import *
import math


def IsPrime(num):
    for i in range(2, int(math.sqrt(num))):
        if num % i == 0:
            return False
    return True


def solution(nums):
    arr = list(sum(list(i)) for i in combinations(nums, 3))
    count = 0
    for i in arr:
        if IsPrime(i):
            count += 1
    return count

File: level_1/소수 만들기.java
class Solution {
    public int solution(int[] nums) {
        int answer = 0;
        for (int i = 0; i < nums.length - 2; i++) {
            for (int j = i + 1; j < nums.length - 1; j++) {
                for (int k = j + 1; k < nums.length; k++) {
                    boolean isPrime = true;
                    int num = nums[i] + nums[j] + nums[k];
                    for (int l = 2; l <= Math.sqrt(num); l++) {
                        if (num % l == 0) {
                            isPrime = false;
                            break;
                        }
                    }
                    if (isPrime)
                        answer++;
                }
            }
        }

        return answer;
    }
}